language-server/src/server.js

```diff
@@ -119,6 +121,44 @@ connection.onInitialized(async () => {
   await validateWorkspace({ changes: [] });
 });
 
+connection.onHover(async (textDocumentPositionParams) => {
+  const { textDocument: { uri: textDocumentURI }, position } = textDocumentPositionParams;
+  const document = documents.get(textDocumentURI);
+  const instance = JsoncInstance.fromTextDocument(document);
+  if (instance.typeOf() === "undefined") {
+    return;
+  }
+  const $schema = instance.get("#/$schema");
+  const contextDialectUri = $schema.value();
+
+  const schemaResources = decomposeSchemaDocument(instance, contextDialectUri);
+  for (const { dialectUri, schemaInstance } of schemaResources) {
+    if (!hasDialect(dialectUri)) {
+      continue;
+    }
+    const [, annotations] = await validate(dialectUri, schemaInstance);
+    const keyword = annotations.keyAtPosition(position);
+    if (keyword.typeOf() !== "undefined") {
+      // Found
+      const description = keyword.annotation("description", dialectUri).join("\n");
+      return buildHover(MarkupKind.Markdown, description, keyword.startPosition(), keyword.endPosition());
+    } else if (keyword.typeOf() === "undefined" && typeof keyword.pointer !== "undefined") {
+      // Found but not a key
+      return;
```

Comment:
Are you sure returning here is the right thing? The node might be in any of the schema resources. If you return when it's not in one, it will never check the others.